### PR TITLE
feat: relax bucket requirement on client metrics

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

When impact metrics are used but no impression events are captured client metrics will have impact metrics sent over the wire but some SDKs will send null bucket. This PR makes our schema more forgiving.

I'm also considering adding impact metrics validation in openapi because today we only do this in joi (I guess we gave impact metrics enough time to stabilize and be part of official openapi schema)

### CODE REVIEW

File: src/lib/features/metrics/impact/impact-metrics.e2e.test.ts [hunk 1]
```diff
@@ -20,11 +20,6 @@ const sendImpactMetrics = async (
         .send({
             appName: 'impact-metrics-app',
             instanceId: 'instance-id',
-            bucket: {
```
Comment (kwasniew) [inline src/lib/features/metrics/impact/impact-metrics.e2e.test.ts:23]: some SDKs send dummy bucket, some don't

File: src/lib/features/metrics/impact/impact-metrics.e2e.test.ts [hunk 2]
```diff
@@ -20,11 +20,7 @@ const sendImpactMetrics = async (
...
-                start: Date.now(),
-                stop: Date.now(),
-                toggles: {},
-            },
+            bucket: null,
```
Comment (kwasniew) [inline src/lib/features/metrics/impact/impact-metrics.e2e.test.ts]: simulating what python sdk is doing

File: src/lib/openapi/spec/client-metrics-schema.ts [hunk 3]
```diff
@@ -59,9 +59,10 @@ export const clientMetricsSchema = {
         },
         bucket: {
             type: 'object',
+            nullable: true,
```
Comment (kwasniew) [inline src/lib/openapi/spec/client-metrics-schema.ts:62]: both null and undefined are accepted now

File: src/test/e2e/api/client/metrics.e2e.test.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ test('should require valid send metrics', async () => {
     return app.request
         .post('/api/client/metrics')
         .send({
-            appName: 'test',
+            instanceId: 'test',
```
Comment (kwasniew) [inline src/test/e2e/api/client/metrics.e2e.test.ts:50]: appName is not valid schema and the only required field so changed it to another field

File: src/lib/features/metrics/impact/impact-metrics.e2e.test.ts (hunk 2 above)
Comment (sighphyre) [reply src/lib/features/metrics/impact/impact-metrics.e2e.test.ts]: Could we do do both - maybe pull this into its own test rather than patch? I have a feeling this case was catching older legacy SDKs that sometimes send start/stop but don't send toggles and that's also important - it's really the wild west out there in SDK land and ideally I'd like tests that prove we're VERY liberal

File: src/lib/features/metrics/client-metrics/metrics-service-v2.ts [hunk 5]
```diff
@@ -285,16 +285,22 @@ export default class ClientMetricsServiceV2 {
...
+            return;
+        }
+
         const clientMetrics: IClientMetricsEnv[] = Object.keys(
-            value.bucket.toggles,
+            bucket.toggles,
         ).map((name) => ({
             featureName: name,
             appName: value.appName,
             environment: value.environment ?? 'default',
```
Comment (sighphyre) [inline src/lib/features/metrics/client-metrics/metrics-service-v2.ts:299]: Huh. That seems weird and dangerous. Just thinking out loud, I know this isn't a thing this PR has changed. Just adding it to my notes 😅

File: src/lib/features/metrics/shared/schema.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ export type ValidatedClientMetrics = {
     environment?: string;
```
Comment (sighphyre) [inline src/lib/features/metrics/shared/schema.ts:19]: Another thinking out loud think for later - we can definitely derive this from the token if it isn't already done. I might need to go double check how this flow is working across the SDKs

**Review by sighphyre — APPROVED** [review]
Yeah awesome, thanks for making this patch. I think this looks great. Usual thoughts but up to you if you want to do something about them, I think this stands as it is

File: src/lib/features/metrics/impact/impact-metrics.e2e.test.ts (hunk 2 above)
Comment (kwasniew) [reply src/lib/features/metrics/impact/impact-metrics.e2e.test.ts]: yes, created a separate test to cover all bucket corner cases
